Some Relion 3.1 fixes

delarosatrevin opened the pull request:
Bug fixes and features:

- [x] * Fixed matching between particles and movies by using micName in polishing
- [x] * Fixed generation of the local motion table when needed (e.g when only some movies failed to align)
- [x] * Allowed to display ctf-refinement results with matplotlib (still need to find the proper color map)
- [x] * Fixed function splitCTFGroups, now replaced with class DefocusGroups and not using Xmipp metadata 
- [x] * Now use proper pixelSize label depending on input set

azazellochg commented on relion/viewers/viewer_ctfrefine.py:
```diff
@@ -170,23 +175,47 @@ def _displayDefocus(self, e=None):
+            return [DataView(img) for img in imgs]
+
+    def _showImagesMatplotlib(self, title, *imgs):
+        import mrcfile
```
for this we'll need to include it into pip requirements?

delarosatrevin replied:
I wasn't sure if we were already using mrcfile somewhere...we could also read the image with ImageHandler class (xmipp under the hood) and use the getData() that will give us the numpy array...probably the best solution right now....anyway this import should not stay there.

azazellochg replied:
it's still there.. :)

delarosatrevin replied:
Yes, I will change this, please keep checking and I will fix this before the merge.

delarosatrevin replied:
Done now with ImageHandler

delarosatrevin commented:
@azazellochg Can you take a look at the PR?

azazellochg commented:
Looking now

azazellochg commented on relion/protocols/protocol_base.py:
```diff
@@ -30,6 +30,7 @@
 from collections import OrderedDict
 from emtable import Table
 
+from pyworkflow.object import String
```
Didn't find where this is used

delarosatrevin replied:
Removed

azazellochg commented on relion/convert/convert_coordinates.py:
```diff
@@ -0,0 +1,239 @@
+# **************************************************************************
```
Am I right that this is just moved code? Nothing new?

delarosatrevin replied:
Yes, moved code from convert_deprecated mainly...there were a lot of coordinates related functions that might help to have in its own implementation file.

azazellochg commented on relion/protocols/protocol_base.py:
```diff
@@ -991,7 +993,14 @@ def _setComputeArgs(self, args):
 
         if self.doGpu:
-            args['--gpu'] = self.gpusToUse.get()
+            gpuStr = self.gpusToUse.get().strip()
```
I didn't have any problems running relion without quotes, but why not

delarosatrevin replied:
I can't remember exactly the case, but Dustin had some issue, not  sure if using :  or spaces

azazellochg commented:
@delarosatrevin I have run ref3d with grouping by defocus. Input star has groups, but output particles set does not have rlnGroupName or number

delarosatrevin commented:
@azazellochg I have addressed your comments, still, the reconstruct test is failing, but I will suggest leaving out of this PR. I asked Marabini help for that.
